traffic_ops/testing/api/v3/cachegroups_test.go

```diff
@@ -16,144 +16,241 @@ package v3
 */
 
 import (
-	"fmt"
+	"encoding/json"
 	"net/http"
 	"net/url"
-	"reflect"
+	"strconv"
 	"testing"
 	"time"
 
 	"github.com/apache/trafficcontrol/lib/go-rfc"
 	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/traffic_ops/testing/api/assert"
+	"github.com/apache/trafficcontrol/traffic_ops/testing/api/utils"
+	"github.com/apache/trafficcontrol/traffic_ops/toclientlib"
+	client "github.com/apache/trafficcontrol/traffic_ops/v3-client"
 )
 
 func TestCacheGroups(t *testing.T) {
 	WithObjs(t, []TCObj{Types, Parameters, CacheGroups, CDNs, Profiles, Statuses, Divisions, Regions, PhysLocations, Servers, Topologies}, func() {
-		GetTestCacheGroupsIMS(t)
-		GetTestCacheGroupsByNameIMS(t)
-		GetTestCacheGroupsByShortNameIMS(t)
-		GetTestCacheGroups(t)
-		GetTestCacheGroupsByName(t)
-		GetTestCacheGroupsByShortName(t)
-		GetTestCacheGroupsByTopology(t)
-		CheckCacheGroupsAuthentication(t)
+
+		tomorrow := time.Now().AddDate(0, 0, 1).Format(time.RFC1123)
 		currentTime := time.Now().UTC().Add(-5 * time.Second)
-		time := currentTime.Format(time.RFC1123)
-		var header http.Header
-		header = make(map[string][]string)
-		header.Set(rfc.IfModifiedSince, time)
-		header.Set(rfc.IfUnmodifiedSince, time)
-		UpdateTestCacheGroups(t)
-		UpdateTestCacheGroupsWithHeaders(t, header)
-		GetTestCacheGroupsAfterChangeIMS(t, header)
-		header = make(map[string][]string)
-		etag := rfc.ETag(currentTime)
-		header.Set(rfc.IfMatch, etag)
-		UpdateTestCacheGroupsWithHeaders(t, header)
+		currentTimeRFC := currentTime.Format(time.RFC1123)
+
+		methodTests := map[string]map[string]struct {
+			endpointId     func() int
+			clientSession  *client.Session
+			requestParams  url.Values
+			requestHeaders http.Header
+			requestBody    map[string]interface{}
+			expectations   []utils.CkReqFunc
+		}{
```

Comment:
This anonymous struct is now in API v3 and API v4. It's fine for now, but if you think it will be used more times, it may be useful to declare it somewhere for reuse.